clusterloader2/docs/network-perf-testcase-enhancement/README.md

```diff
@@ -0,0 +1,286 @@
+	
+      2.2.2 Client-Server POD on different worker node (N:M) (TCP/UDP traffic)
+	  
+	* Client POD size N=50 and Server POD size M=50. Each server pod will be behind one unique ClusterIP/NodePort Svc. So 50 clusterIP will be created for TCP/UDP traffic only. ( 1 Client POD ---> 1 ClusterIP ---> 1 Server POD)
+	* This test will require 100 worker nodes i.e N+M, so that we can make sure client and server pods are not co-located.
+	* Each metric will be a percentile of 50 pairs.
+		
+     TestcaseId  | Testcase Description | Tool | Metrics Measured
+     ----------- | ---------------------|------|-----------------
+     TC2.2.2.1  | Send TCP traffic from 1 client pod IPs to 1 server pod is located behind respective unique ClusterIP (1:1) | iperf3 | Throughput in Gbits/s
+     TC2.2.2.2  | Send UDP traffic from 1 client pod IPs to 1 server pods is located behind respective unique ClusterIP (1:1) | iperf2 | Packets per second (pps), Loss Rate, Latency, jitter
+     TC2.2.2.3  | Send TCP traffic from 1 client pod IPs to NodePort svc IP. 1 server pod is located behind NodePort service (1:1) | iperf3 | Throughput in Gbits/s
+     TC2.2.2.4  | Send UDP traffic from 1 client pod IPs to NodePort svc IP. 1 server pod is located behind NodePort service (1:1) | iperf2 | Packets per second (pps), Loss Rate, Latency, jitter
+    
+
+### Tools
+
+We intend to use the following tools for testing :
+
+1. iperf3: For TCP throughput measurement.
+
+2. iperf2: For UDP latency, loss rate , pps measurement.
+
+3. Siege: Siege is an open source benchmark utility. 
+   It will be used for measuring HTTP/HTTPs metrics such as total number of hits recorded, 
+   bytes transferred, response time, concurrency, and return status. 
+
+Other tools such as Trex will be used in future for further enhancements.
+
+### Resource Requirements
+
+1. Worker nodes : We will need worker nodes (VMs) to run client and server pods.
+   As mentioned in test scenarios, we require 100 worker nodes to run the testcases.
+
+2. Memory : Each worker node should have atleast 4 Gb of memory. 
+
+3. vCPUs : Each worker node should have atleast 2 vCPUs to run tools at full capacity.
+   We can pin the vCPUs to tools to run them efficienlty. If multiple client-server pods 
+   are run on same worker node, then more then 2 vCPUs might be required.
+
+### Future Enhancements
+
+Performance benchmarking frameworks can be enhanced to include testcases related to various CNIs.
+
+### Benefits
+
+1) Our proposal attempts to standardize the test scenarios and cover most common performance metrics.
+2) Detecting performance degradation in early phase of feature delivery.
+3) With standard test scenarios available, kubernetes users will be able adopt these test frameworks
+   to validate their downstream offerings.  
+   
+## Design Details
+
+
+Kubernetes uses the Prow framework for CI/CD and the new network performance related tests are also planned to be added as part of prow jobs.
```

Comment:
nit: can you please split the lines in this section to make them shorter?

Reply:
yes i have made the changes.